recipes/libsodium/1.0.18/conanfile.py

```diff
@@ -0,0 +1,223 @@
+        if self.settings.arch.value == "x86":
+            return "Win32"
+        return "x64"
+
+    @property
+    def _vs_configuration(self):
+        configuration = ""
+        if self.options.shared:
+            configuration += "Dyn"
+        else:
+            configuration += "Static"
+        build_type = str(self.settings.build_type)
+        if build_type == "RelWithDebInfo":
+            build_type = "Release"
+        configuration += build_type
+        return configuration
+
+    @property
+    def _vs_sln_folder(self):
+        if self.settings.compiler.version == "14":
+            return "vs2015"
+        elif self.settings.compiler.version == "15":
+            return "vs2017"
+        elif self.settings.compiler.version == "16":
+            return "vs2019"
+        else:
+            raise ConanInvalidConfiguration("Unsupported msvc version: {}".format(self.settings.compiler.version))
+
+    @property
+    def _runtime_prefix(self):
+        return str(self.settings.compiler.runtime)[:2]
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _build_visual(self):
+        sln_path = os.path.join(self.build_folder, self._source_subfolder, "builds", "msvc", self._vs_sln_folder, "libsodium.sln")
+
+        msbuild = MSBuild(self)
+        msbuild.build(sln_path, upgrade_project=False, platforms={"x86": "Win32"}, build_type=self._vs_configuration)
+
+    def _build_autotools_impl(self, configure_args):
+        win_bash = False
+        if self._is_mingw:
+            win_bash = True
+
+        autotools = AutoToolsBuildEnvironment(self, win_bash=win_bash)
+        if self._is_mingw:
+            self.run("autoreconf -i", cwd=self._source_subfolder, win_bash=win_bash)
+        autotools.configure(args=configure_args, configure_dir=self._source_subfolder, host=False)
+        autotools.make(args=["-j%s" % str(tools.cpu_count())])
+        autotools.make(target="install")
+
+    def _build_autotools_linux(self, configure_args):
+        self._build_autotools_impl(configure_args)
+
+    def _build_autotools_emscripten(self, configure_args):
+        self.run("./dist-build/emscripten.sh --standard", cwd=self._source_subfolder)
+
+    def _build_autotools_android(self, configure_args):
+        host_arch = "%s-linux-%s" % (self._arch_id_str_compiler, self._android_id_str)
+        configure_args.append("--host=%s" % host_arch)
+        self._build_autotools_impl(configure_args)
+
+    def _build_autotools_mingw(self, configure_args):
+        if self.settings.arch == "x86":
+            arch = "i686"
+        else:
+            arch = "x86_64"
```

Comment:
I think smething like this could be valid:
`arch = "i686" if self.settings.arch == "x86" else self.settings.arch`